### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by MDLC01**
I changed the chessboard example of [the documentation for `grid.cell`](https://typst.app/docs/reference/layout/grid/#definitions-cell) so that the bottom right square is white.

I also clarified that [`array.insert`](https://typst.app/docs/reference/foundations/array/#definitions-insert) shifts subsequent elements to the right.

### CODE REVIEW

File: crates/typst/src/layout/grid/mod.rs [hunk 1]
```diff
@@ -701,15 +701,15 @@ pub struct GridVLine {
...
+///     if calc.odd(x + y) { "7F8396" }
+///     else { "EFF0F3" }
 ///   ),
 ///   columns: (1em,) * 8,
 ///   rows: 1em,
 ///   align: center + horizon,
 ///
 ///   [♜], [♞], [♝], [♛], [♚], [♝], [♞], [♜],
-///   [♟], [♟], [♟], [♟], [],  [♟], [♟], [♟],
+///   [♟], [♟], [♟], [♟], [],   [♟], [♟], [♟],
```
Comment (PgBiel) [inline crates/typst/src/layout/grid/mod.rs:712]: This seems to have undone the alignment in the source code.

```suggestion
///   [♟], [♟], [♟], [♟], [],  [♟], [♟], [♟],
```

Comment (MDLC01) [reply crates/typst/src/layout/grid/mod.rs:712]: Oooh I see. This is because chess pieces are not displayed with a monospace font on my end.

**Comment by PgBiel** [comment]
LGTM :+1:

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**